schema.json

```diff
@@ -29,7 +29,12 @@
     },
     "type": "object",
     "properties": {
+        "$id": {
+            "type": "string",
+            "format": "uriref"
+        },
         "id": {
+            "description": "This keyword has been deprecated in favor of \"$id\"",
```

Comment:
I don't have strong feelings on this.  The deprecation option came in response to migration concerns, although as you note there are several easy options for handling the change.  If y'all decide to drop `id` entirely I'll be happy to update the PR.